inference-engine/src/transformations/src/transformations/common_optimizations/pad_fusion.cpp

```diff
@@ -0,0 +1,202 @@
+        auto max_pool = std::dynamic_pointer_cast<opset5::MaxPool>(pattern_map[max_pool_pattern].get_node_shared_ptr());
+        if (!can_be_fused(pad, max_pool, pad_value_const, pads_begin, pads_end))
+            return false;
+
+        Shape new_pads_begin, new_pads_end;
+        std::tie(new_pads_begin, new_pads_end) = new_pad_values(pads_begin, pads_end, max_pool);
+        auto new_max_pool = std::make_shared<opset5::MaxPool>(data, max_pool->get_strides(),
+                                                              new_pads_begin, new_pads_end,
+                                                              max_pool->get_kernel(), max_pool->get_rounding_type(),
+                                                              op::PadType::EXPLICIT);
+        new_max_pool->set_friendly_name(max_pool->get_friendly_name());
+
+        copy_runtime_info(max_pool, new_max_pool);
+        replace_node(max_pool, new_max_pool);
+
+        return true;
+    };
+
+    auto m = std::make_shared<pattern::Matcher>(max_pool_pattern, "PadFusionMaxPool");
+    this->register_matcher(m, callback);
+}
+
+NGRAPH_RTTI_DEFINITION(pass::PadFusionConvolution, "PadFusionConvolution", 0);
+
+static std::tuple<CoordinateDiff, CoordinateDiff> new_pad_values(const std::shared_ptr<opset5::Constant>& pads_begin,
+                                                                 const std::shared_ptr<opset5::Constant>& pads_end,
+                                                                 const std::shared_ptr<opset5::Convolution>& node) {
+    auto node_pads_begin = node->get_pads_begin();
+    auto node_pads_end = node->get_pads_end();
+    auto pads_begin_val = pads_begin->cast_vector<size_t>();
+    auto pads_end_val = pads_end->cast_vector<size_t>();
+
+    std::transform(node_pads_begin.begin(), node_pads_begin.end(),
+                   pads_begin_val.begin() + 2, node_pads_begin.begin(),
+                   [] (std::ptrdiff_t a, size_t b) -> std::ptrdiff_t { return a + b; });
+    std::transform(node_pads_end.begin(), node_pads_end.end(),
+                   pads_end_val.begin() + 2, node_pads_end.begin(),
+                   [] (std::ptrdiff_t a, size_t b) -> std::ptrdiff_t { return a + b; });
+
+    return std::make_tuple(node_pads_begin, node_pads_end);
+}
+
+
+
+pass::PadFusionConvolution::PadFusionConvolution() {
+    auto data_pattern = pattern::any_input();
+    auto filter_pattern = pattern::any_input();
+    auto pads_begin_pattern = pattern::wrap_type<opset5::Constant>();
+    auto pads_end_pattern = pattern::wrap_type<opset5::Constant>();
+    auto pad_value_pattern = pattern::wrap_type<opset5::Constant>();
+    auto pad_node_pattern = pattern::wrap_type<opset5::Pad>({data_pattern, pads_begin_pattern, pads_end_pattern, pad_value_pattern});
+    auto conv_pattern = pattern::wrap_type<opset5::Convolution>({pad_node_pattern, filter_pattern});
```

Comment:
I think we need to covert all convolution types then:
* Convolution
* GroupConvolution
* ConvolutionBackPropData
* GroupConvolutionBackPropData